reana_client/utils.py

```diff
@@ -425,3 +441,28 @@ def run_command(cmd, display=True, return_output=False, stderr_output=False):
         if stderr_output:
             sys.exit(err.output.decode())
         sys.exit(err.returncode)
+
+
+def _validate_workspace(workspace, access_token):
+    """Validate workspace in REANA specification file.
+
+    :param workspace: workspace path to be validated.
+    :param access_token: access token of the current user.
+
+    :raises ValidationError: Given workspace in REANA spec file does not validate against
+        allowed workspaces.
+    """
+    from reana_client.api.client import workspaces
+
+    available_workspaces = workspaces(access_token).get("workspaces_available", [])
+    try:
```

Comment:
Perhaps here we could reuse [validate_workspace()](https://github.com/reanahub/reana-commons/pull/286/files#diff-0e9c97b2308d9b723f7552586946c165404f92f2ca51ed311d00f6159dd26514R14) from `reana-commons` and modify it to optionally accept workspace path list? Something like `validate_workspace(workspace_option, paths=WORKSPACE_PATHS)`. In that case we would have a central validation place where we pass the list from `reana-client` and and use the envar from `r-server`. WDYT?